### PR TITLE
Sync Formik ActionBlock

### LINKED ISSUE #52: [Syncing] formik/ActionBlock
## Points to note
- Component in action - `formik/ActionBlock.js`
- Github Urls to this component:
  * [neeto-ui/formik/ActionBlock.js](https://github.com/bigbinary/neeto-ui/tree/67ee4a5188c5feccbeca2767107651fb24932764/lib/components/formik/ActionBlock.js)
  * [NitroUI/Formik/ActionBlock.js](https://github.com/bigbinary/neeto-help-web/tree/0b11a7b0f98ac7c533df1e9ec37cf900b4cbc901/app/javascript/Admin/Components/NitroUI/Formik/ActionBlock.js)
- :green_circle: Green color signifies how that particular line is written in `neeto-ui` package
- :red_circle: Red color signifies how that particular line is written in `NitroUI` internals

## Sync Required or Not
- Are there significant changes for `formik/ActionBlock.js` that needs to be synced to `neeto-ui` package from `NitroUI`? - **Yes**
- Will bring in the `dataTestId` attr

## Diff
```diff
--- NitroUI/formik/ActionBlock.js	2020-11-28 16:54:23.922371400 +0530
+++ neetoui-official/formik/ActionBlock.js	2020-11-28 16:49:18.765840500 +0530
@@ -1,5 +1,5 @@
 import React, { useEffect } from "react";
-import { Button } from "NitroUI";
+import Button from "../Button";
 import { useFormikContext } from "formik";
 
 const ActionBlock = () => {
@@ -11,7 +11,6 @@
         label="Reset changes"
         className="mr-4"
         onClick={handleReset}
-        dataTestId="reset-changes-button"
       />
       <Button
         style="primary"
@@ -19,7 +18,6 @@
         icon="ri-save-3-fill"
         loading={isSubmitting}
         onClick={handleSubmit}
-        dataTestId="save-changes-button"
       />
     </div>
   );
```

### PR DESCRIPTION
**Pull request by yedhink**
Fixes #52 

@goutham-subramanyam _a

Since I  am not that familiar with the use case of `formik/ActionBlock`, I'm not sure what tests I needed to do. But I have added an example invocation in `src/index.js`. So please do review and merge if everything looks good.

Also tried to future-proof `data-test-id` based on https://github.com/bigbinary/neeto-ui/pull/70#discussion_r537541501

cc: @edwinbbu Let me know if you see any anti-patterns. Thanks.

### CODE REVIEW

**Comment by yedhink** [comment]
@edwinbbu 
@goutham-subramanyam 

**If merging this issue could be prioritized, then it would be very helpful.** I had added a test scenario using formik. I'd have to repeat it in all other `formik` related syncing changes, before working on them, if this PR doesn't get merged soon. In order to avoid that, if we prioritize this issue, then it would be much appreciated.

Thanks.

File: lib/components/formik/ActionBlock.js [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
-const ActionBlock = () => {
+
+const ActionBlock = props => {
   const { handleReset, handleSubmit, isSubmitting } = useFormikContext();
+
   return (
     <div className="flex flex-row items-center justify-end">
       <Button
         style="secondary"
         label="Reset changes"
```
Comment (goutham-subramanyam) [inline lib/components/formik/ActionBlock.js:13]: `
```suggestion
        label="Reset Changes"
```

File: lib/components/formik/ActionBlock.js [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
         className="mr-4"
         onClick={handleReset}
+        dataTestId={props["reset-data-test-id"] || "reset-changes-button"}
```
Comment (goutham-subramanyam) [inline lib/components/formik/ActionBlock.js:16]: We should be passing it as `data-test-id`

**Review by goutham-subramanyam — COMMENTED** [review]
@yedhink _a

File: lib/components/formik/ActionBlock.js (hunk 2 above)
Comment (yedhink) [reply lib/components/formik/ActionBlock.js:16]: @goutham-subramanyam _a

So what about the `data-test-id` for next save Button? Can both the buttons have the same `data-test-id` value? In neeto-help it's not like that. Thus two different prop keys.

Please confirm whether you want same `data-test-id` for both the Buttons. Thanks

Comment (goutham-subramanyam) [reply lib/components/formik/ActionBlock.js:16]: @yedhink 

```suggestion
        data-test-id={props["reset-data-test-id"] || "reset-changes-button"}
```

**Comment by yedhink** [comment]
@goutham-subramanyam _a

I have made the changes. Please review.

**Review by goutham-subramanyam — APPROVED** [review]
LGTM

**Pull request merged**